qiskit/providers/ibmq/runtime/runtime_program.py

```diff
@@ -255,3 +258,77 @@ class ProgramResult(NamedTuple):
     name: str
     description: str
     type: str
+
+class ParameterNamespace(SimpleNamespace):
+    """ An abstraction for SimpleNamespace that offers param validation.
+
+    Note: Unused/Unnecessary params does not cause error (still validates)
+
+    Args:
+        params (List[ProgramParameter]): The program's input parameters
+
+    Raises
+        IBMQInputValueException on validation fail
+    """
+
+    def __init__(self, params: List[ProgramParameter]):
+        super().__init__()
+        # Allow access to the raw program parameters list
+        self.params_list = params
+        # For localized logic, create store of parameters in dictionary
+        self.__program_params: dict = {}
+
+        for param in params:
+            # (1) Add parameters to a dict by name
+            setattr(self, param.name, None)
+            # (2) Store the program params for validation
+            self.__program_params[param.name] = param
+
+    def validate(self) -> None:
+        """Validates the user's usage of the program's inputs
+
+        Raises:
+            IBMQInputValueError if validation fails
+        """
+
+        # Iterate through the user's stored inputs
+        for param_name, program_param in self.__program_params.items():
+            # Set invariants: User-specified parameter value (value) and whether it's required (req)
+            value = None
+            req = program_param.required
+            # Check there exists a program parameter of that name.
+            try:
+                # Get the value currently stored for this parameter (user-adjusted)
+                value = self.__dict__[param_name]
+            except AttributeError:
+                if req:
+                    raise IBMQInputValueError('Param (%s) missing required value!' % param_name)
+            # Check for empty value
+            if not value or value is None:
+                if req:
+                    raise IBMQInputValueError('Param (%s) missing required value!' % param_name)
```

Comment:
You can combine the two with `getattr(self, param_name, None)`.

Reply:
Fixed!

Reply:
I don't think this is fixed, unless you forgot to push lol